src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,181 @@
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public virtual IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+
+            return ToHtmlString(new LocalizedString(key, EncodeArguments(stringValue, arguments)));
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected virtual LocalizedHtmlString ToHtmlString(LocalizedString result) =>
+            new LocalizedHtmlString(result.Name, result.Value, result.ResourceNotFound);
+
+        /// <summary>
+        /// Encodes the arguments based on the object type.
+        /// </summary>
+        /// <param name="resourceString">The resourceString whose arguments need to be encoded.</param>
+        /// <param name="arguments">The array of objects to encode.</param>
+        /// <returns>The string with encoded arguments.</returns>
+        protected virtual string EncodeArguments([NotNull] string resourceString, [NotNull] object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            char currentCharacter;
+            StringBuilder tokenBuffer = null;
+            var outputBuffer = new StringBuilder();
+            var isToken = false;
+
+            while (position < length)
+            {
+                currentCharacter = resourceString[position];
+
+                position++;
+                if (currentCharacter == '}')
+                {
+                    tokenBuffer.Append(currentCharacter);
+
+                    // Treat as escape character for }}
+                    while (position < length && resourceString[position] == '}') 
+                    {
+                        currentCharacter = resourceString[position];
+                        tokenBuffer.Append(currentCharacter);
+                        position++;
+                    }
+
+                    if (tokenBuffer != null && tokenBuffer.Length > 0)
+                    {
+                        var formattedString = string.Format(tokenBuffer.ToString(), arguments);
+                        outputBuffer.Append(EncodeArgument(formattedString));
```

Comment:
Should have mentioned earlier: Given a resource such as "{{should not be encoded}} is not encoded {0}", the "{{should not be encoded}}" bit _will_ be HTML encoded. This could result in double-encoding and is inconsistent with the "just encodes the arguments" expectation. Either should not put "{{" or "}}" into the token buffer or should `Append()` the formatted content of `tokenBuffer` into `outputBuffer` as soon as it contains two curlies -- without encoding.